nemo_text_processing/text_normalization/zh/taggers/cardinal.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pynini
+from nemo_text_processing.text_normalization.zh.graph_utils import GraphFst
+from nemo_text_processing.text_normalization.zh.utils import (
+    UNIT_1e01,
+    UNIT_1e02,
+    UNIT_1e03,
+    UNIT_1e04,
+    UNIT_1e08,
+    UNIT_1e12,
+    get_abs_path,
+)
+from pynini.lib import pynutil
+
+
+class Cardinal(GraphFst):
+    '''
+        self.graph_cardinal:
+        5       -> cardinal{ integer: "五"}
+        12      -> cardinal{ integer: "十二"}
+        213     -> cardinal{ integer: "二百一十三"}
+        3123    -> cardinal{ integer: "三千一百二十三"}
+        3,123   -> cardinal{ integer: "三千一百二十三"}
+        51234   -> cardinal{ integer: "五万一千二百三十四"}
+        51,234  -> cardinal{ integer: "五万一千二百三十四"}
+        0.125   -> cardinal{ integer: "零点一二五"}
+        self.fst:
+        123     -> cardinal{ integer: "一二三" }
```

Comment:
Thank you for your advice！The opinion I  handle with number is:①when there is a unit,we read it into cardinal forms,just like "20头牛、125个人、3000元",so the measure part is used to normalize number with unit both in English and Chinese.②When it comes to a single number, just like  "747 or 10086"only, although we prefer reading it to"七百四十七 or 一万零八十六", in real cases,the correct forms may be “波音七四七、编号七四七人员、拨打一二三零六".
To put it simply，the previous work just transport number into cardinal forms and I just found it has a lower accuracy in lots of real cases we tested.Therefore,to handle with more detailed cases,I set it more detailed,just like now.I think we could not deal with all the cases by one form,and the present forms may lead to more accuracy in Chinese Text Normalization.

Reply:
Thank you @mzxcpp  for the explanation. Very interesting observation. @BuyuanCui, did you see similar things like this and can confirm this?
@mzxcpp is it possible to share the real cases you tested on? that would be highly interesting.

Reply:
Thank you for taking my idea into consideration!I just base on real cases in Leaderboard.
https://github.com/SpeechColab/Leaderboard
and
https://github.com/dophist/textnorm/blob/main/zh/misc/speechio_0_15.tsv
and I have just gone through some cases and find the following cases:

原知识产权5条热线和平台的整合，实现12315一个号码对外一个平台受理，方便消费者。
针对禁止报出的波音737ng型飞机被发现了机身出现裂缝一事。美国波音公司10月31日宣布。
在5个月内连续发生两次空难后，目前波音公司已暂停向客户交付t37-8及t37-9新飞机。
当天在伊朗首都德黑兰附近机场失事的波音707货机的黑匣子已经找到。
春节期间，各界群众若发现涉黄涉非违法行为，可通过全国扫黄打非办公室举报电话12390。
鲁豫探访奇虎360，带您走进这位互联网传奇老炮周弘毅。
2002年一度领先市场份额的3721逐渐迎来强劲对手。
感谢3158创业信息网的大力支持。
您这个是号是000~008。
身高1米84的克洛泽似乎是在有意避开人们对他只会投球的质疑。
最近呢有个小朋友跟我说，呢他们公司准备要开展996工作制，就是早上9点工作到晚上9点。
1234这4个队员会选择进攻。这样一来先防守他们。
510有点有一点点的偏局红调唉510我试一下吧510应该还可以。我试一下510好不好？
DC37烟熏玫瑰色，所有女生都可以用的颜色。

These cases will go wrong if we set each number into cardinal forms.And it's difficult to try to cover all the special forms with number string.I think it may be easier to cover all the common units rather than all this special forms.

Admittedly,there are cases with wrong answer:

一年每年呢给你免24,000，如果你是非独生子女。
9999995这么大。

In these cases,speaker ignores the unit because in some situation,others can know the correct meaning in Chinese,just like "元" may be left out.
Just my experimental idea and I will follow your ideas whether to set it as cardinal forms or keep the current forms.